sdk/tests/conformance2/misc/uninitialized-test-2.html

```diff
@@ -72,68 +95,208 @@
     return texture;
 }
 
-function checkNonZeroPixels(texture, target, texWidth, texHeight, level, layer) {
-    gl.bindTexture(target, null);
+function checkNonZeroPixels(texture, target, format, type, texWidth, texHeight, level, layer, exceptions) {
+    var is3d = (target == gl.TEXTURE_3D || target == gl.TEXTURE_2D_ARRAY);
+    switch (target) {
+      case gl.TEXTURE_CUBE_MAP_POSITIVE_X:
+      case gl.TEXTURE_CUBE_MAP_NEGATIVE_X:
+      case gl.TEXTURE_CUBE_MAP_POSITIVE_Y:
+      case gl.TEXTURE_CUBE_MAP_NEGATIVE_Y:
+      case gl.TEXTURE_CUBE_MAP_POSITIVE_Z:
+      case gl.TEXTURE_CUBE_MAP_NEGATIVE_Z:
+        gl.bindTexture(gl.TEXTURE_CUBE_MAP, null);
+        break;
+      default:
+        gl.bindTexture(target, null);
+        break;
+    }
     var fb = gl.createFramebuffer();
     gl.bindFramebuffer(gl.FRAMEBUFFER, fb);
-    gl.framebufferTextureLayer(gl.FRAMEBUFFER, gl.COLOR_ATTACHMENT0, texture, level, layer);
+    if (is3d) {
+        gl.framebufferTextureLayer(gl.FRAMEBUFFER, gl.COLOR_ATTACHMENT0, texture, level, layer);
+    } else {
+        gl.framebufferTexture2D(gl.FRAMEBUFFER, gl.COLOR_ATTACHMENT0, target, texture, level);
+    }
     shouldBe("gl.checkFramebufferStatus(gl.FRAMEBUFFER)", "gl.FRAMEBUFFER_COMPLETE");
 
-    var data = new Uint8Array(texWidth * texHeight * 4);
-    gl.readPixels(0, 0, texWidth, texHeight, gl.RGBA, gl.UNSIGNED_BYTE, data);
+    var data;
+    switch (type) {
+      case gl.UNSIGNED_INT:
+        data = new Uint32Array(texWidth * texHeight * 4);
+        break;
+      case gl.INT:
+        data = new Int32Array(texWidth * texHeight * 4);
+        break;
+      case gl.UNSIGNED_BYTE:
+      default:
+        data = new Uint8Array(texWidth * texHeight * 4);
+        break;
+    }
+    gl.readPixels(0, 0, texWidth, texHeight, format, type, data);
 
     var k = 0;
+    var failed_exceptions = 0;
     for (var y = 0; y < texHeight; ++y) {
         for (var x = 0; x < texWidth; ++x) {
             var index = (y * texWidth + x) * 4;
+            var is_exception = false;
+            for (var ii = 0; ii < exceptions.length; ++ii) {
+                if (exceptions[ii].x == x && exceptions[ii].y == y) {
+                    is_exception = true;
+                    if (data[index] != exceptions[ii].r ||
+                        data[index + 1] != exceptions[ii].g ||
+                        data[index + 2] != exceptions[ii].b ||
+                        data[index + 3] != exceptions[ii].a) {
```

Comment:
Is a tolerance needed for this comparison? If the back buffer happens to change semantics slightly (for example, it might use an sRGB texture on some implementations) then the read-back pixel may not be exactly what was sampled in the texture.